app/invocation/invocation_action_card.tsx

```diff
@@ -127,65 +128,155 @@ export default class ActionCardComponent extends React.Component<Props, State> {
+                            Output Upload Started @{" "}
+                            {format.formatTimestampSecondsNanos(
+                              this.state.actionResult.executionMetadata.outputUploadStartTimestamp.seconds,
+                              this.state.actionResult.executionMetadata.outputUploadStartTimestamp.nanos
+                            )}
+                          </div>
+                          <div className="metadata-detail">
+                            Output Upload Completed @{" "}
+                            {format.formatTimestampSecondsNanos(
+                              this.state.actionResult.executionMetadata.outputUploadCompletedTimestamp.seconds,
+                              this.state.actionResult.executionMetadata.outputUploadCompletedTimestamp.nanos
+                            )}
+                          </div>
+                          <div className="metadata-info">
+                            Worker Completed @{" "}
+                            {format.formatTimestampSecondsNanos(
+                              this.state.actionResult.executionMetadata.workerCompletedTimestamp.seconds,
+                              this.state.actionResult.executionMetadata.workerCompletedTimestamp.nanos
+                            )}
+                          </div>
+                        </div>
+                      ) : (
+                        <div>None found.</div>
+                      )}
+                    </div>
+                    <div className="action-section">
+                      <div className="action-property-title">Output Files:</div>
+                      {this.state.actionResult.outputFiles ? (
+                        <div className="action-list">
+                          {this.state.actionResult.outputFiles.map((file) => (
+                            <div>
+                              <span className="prop-name">PATH=</span>
+                              <span className="prop-value">{file.path}</span>
+                              <div>
+                                <span className="prop-name">Executable? </span>
+                                <span className="prop-value">{file.isExecutable.toString().toUpperCase()}</span>
+                              </div>
+                            </div>
+                          ))}
+                        </div>
+                      ) : (
+                        <div>None found.</div>
+                      )}
+                    </div>
+                    <div className="action-section">
+                      <div className="action-property-title">Output Directories:</div>
+                      {this.state.actionResult.outputDirectories.length ? (
+                        <div className="action-list">
+                          {this.state.actionResult.outputDirectories.map((file) => (
+                            <div>
+                              <span className="prop-name">PATH=</span>
```

Comment:
Same comment here re. path formatting

Reply:
ah these are directories that don't have the executable information, i renamed file to dir for less confusion